orttraining/orttraining/python/ort_trainer.py

```diff
@@ -711,6 +712,20 @@ def prepare_input_and_fetches(self, input_desc_with_, internal_learning_rate, in
         if internal_loss_scale is not None:
             input = input + (internal_loss_scale,)
 
+        # loss_scale input name is needed to call train_step.
+        # loss_scale input name is determined at the time training session is created.
+        # training session creation is delay to train_step.
+        # as a result, loss_scale input name is not available when calling train_step. 
+        # to work around this problem, we have to use the following pattern matching code to get the loss_scaler argument.
+        # this problem can be solved if leter it is decided that loss scaler is always internal.
+        # note: when self.loss_fn_ is present, following pattern is invalid. 
+        extra_kargs_keys = [key for key in kwargs.keys() if key not in [desc.name_ for desc in input_desc_with_]]
+        if self.loss_fn_ is None and len(input_desc_with_) > len(input) and len(extra_kargs_keys) == 1 and internal_loss_scale is None:
```

Comment:
`len(extra_kargs_keys) == 1` prevents future extensions to this dict. If it has more than one extra arg, extra_kargs_keys[0] may not work because python dict does not guarantee elements ordering